src/providers/node/mod.rs

```diff
@@ -203,9 +219,25 @@ impl Provider for NodeProvider {
         // Start
         let start = NodeProvider::get_start_cmd(app, env)?.map(StartPhase::new);
 
-        let mut plan = BuildPlan::new(&vec![setup, install, build], start);
-        plan.add_variables(NodeProvider::get_node_environment_variables());
+        let mut phases = vec![setup, install, build];
+        if let Some(caddy) = SpaProvider::caddy_phase(app, env) {
+            phases.insert(1, caddy); // insert after setup and before build
+        }
 
+        let mut plan = BuildPlan::new(&phases, start);
+        if SpaProvider::caddy_phase(app, env).is_some() {
+            plan.add_static_assets(static_asset_list! {
+                "Caddyfile" => include_str!("spa/Caddyfile")
```

Comment:
Can the Caddy provider return the static assets that are necessary. That way this top-level node provider doesn't need to know about the internal details of how to build an SPA.